reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/serpapi/SerpApiClient.kt

```diff
@@ -0,0 +1,71 @@
+package com.xebia.functional.xef.reasoning.serpapi
+
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonNames
+
+class SerpApiClient : AutoCloseable {
+
+    private val http: HttpClient = HttpClient {
+        install(HttpTimeout) {
+            requestTimeoutMillis = 60_000
+            connectTimeoutMillis = 60_000
+        }
+        install(HttpRequestRetry)
+        install(ContentNegotiation) {
+            json(
+                Json {
+                    encodeDefaults = false
+                    isLenient = true
+                    ignoreUnknownKeys = true
+                }
+            )
+        }
+    }
+
+    @Serializable
```

Comment:
Does this ever get serialized? It looks like the `SearchData` properties are used to construct a query string manually from its properties, which means this class does not need to be `@Serializable`

Reply:
Done